### PR TITLE
Avoid infinite recursion for matrix rmul

### PR DESCRIPTION
**Pull request by mloubout**
#### Brief description of what is fixed or changed

This small update prevent infinite recursion in some cases for a Matrix `__rmul__`. Currently doing 
```
a  = Matrix(3, 3, rand(3, 3).ravel)
b = MyMatrix(3, 3, rand(3, 3).ravel)
a*b
```
where b is a custom matrix with higher priority than a, the the product produces and infinite recursion since 
`other._new(other.as_mutable() * self) = a._new( a * b) = a*b`. Need to fall back to the actual `evaluate_rmul` the same way
the fallback is `eval_mul` for the `__mul__` case.

#### Other comments

Needs new test

#### Release Notes

* matrices
  * Prevents infinite recursion with rmul

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
Looks good but a test is needed

File: sympy/matrices/tests/test_commonmatrix.py [hunk 1]
```diff
@@ -1036,3 +1036,18 @@ def test_companion():
         Matrix([[0, -c0], [1, -c1]])
     assert Matrix.companion(Poly([1, c2, c1, c0], x)) == \
         Matrix([[0, 0, -c0], [1, 0, -c1], [0, 1, -c2]])
+
+
+def test_rmul_pr19860():
+    class Foo(ImmutableDenseMatrix):
+        _op_priority = 11.01
```
Comment (oscarbenjamin) [inline sympy/matrices/tests/test_commonmatrix.py:1043]: It would be better to use something like `ImmutableDenseMatrix._op_priority + .01`

**Comment by oscarbenjamin** [comment]
There are a lot of `__rmul__` methods in sympy. Does this issue potentially apply to any of the others?

**Comment by mloubout** [comment]
>Does this issue potentially apply to any of the others?

I haven't seen the issue pop up other places no but will double check.

File: sympy/matrices/tests/test_commonmatrix.py (hunk 1 above)
Comment (mloubout) [reply sympy/matrices/tests/test_commonmatrix.py]: Fixed

**Comment by oscarbenjamin** [comment]
Looks good. Thanks!

**Pull request merged**